modules/apps/object/object-storage-salesforce/src/main/java/com/liferay/object/storage/salesforce/internal/odata/filter/expression/SOSQLExpressionVisitorImpl.java

```diff
@@ -127,7 +128,12 @@ public Object visitLiteralExpression(LiteralExpression literalExpression)
 		if (!Objects.equals(
 				LiteralExpression.Type.DATE, literalExpression.getType()) &&
 			!Objects.equals(
-				LiteralExpression.Type.STRING, literalExpression.getType())) {
+				LiteralExpression.Type.STRING, literalExpression.getType()) &&
```

Comment:
@jairmedeiros could you sort it? The following order: BOOLEAN, DATE, DATE_TIME, STRING

Reply:
done!